### PR TITLE
Use agent events in events table

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes a part of #2501

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [ ] Have you checked that you haven't introduced any duplicate code?
    > There is duplication which needs to be refactored

## Testing Checklist

* [ ] ~Added relevant unit tests?~
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running the Island with different dummy events
* [x] If applicable, add screenshots or log transcripts of the feature working

image

### CODE REVIEW

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/monkey_island/cc/ui/src/components/ui-components/EventsTable.js [hunk 1]
```diff
@@ -34,21 +32,113 @@ const table_options = {
...
+  // event_target is a machine ID (positive integer)
+  if ((parseInt(event_target) == event_target) && (event_target > 0)) {
+    for (let machine of machines) {
+      if (event_target === machine['id']) {
+        if ((machine['hostname'] !== null) && (machine['hostname'] !== '')) {
+          return machine['hostname'];
+        }
+        else {
+          return machine['network_interfaces'][0].split('/')[0];
+        }
```
[the remark this answers is not in the document]
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/ui/src/components/ui-components/EventsTable.js:54]: We have another task in the issue to remove the duplication.

File: monkey/monkey_island/cc/ui/src/components/ui-components/EventsTable.js [hunk 2]
```diff
@@ -34,21 +32,113 @@ const table_options = {
...
+  for (let agent of agents) {
+    if (event_source === agent['id']) {
+      for (let machine of machines) {
+        if (agent['machine_id'] === machine['id']) {
+          if ((machine['hostname'] !== null) && (machine['hostname'] !== '')) {
+            hostname = machine['hostname'];
+          }
+          else {
+            hostname = machine['network_interfaces'][0].split('/')[0];
+          }
```
[the remark this answers is not in the document]
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/ui/src/components/ui-components/EventsTable.js:110]: We have another task in the issue to remove the duplication.

File: monkey/common/agent_events/agent_shutdown_event.py [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
+
 from . import AbstractAgentEvent
 
 
 class AgentShutdownEvent(AbstractAgentEvent):
     """
     An event that occurs when the Agent shuts down
     """
+
+    tags: FrozenSet[str] = Field(default=frozenset({"agent-shutdown"}))
```
[the remark this answers is not in the document]
Comment (ilija-lazoroski) [reply monkey/common/agent_events/agent_shutdown_event.py:13]: Probably we though that all events should have tags... Though now it seems like it is not needed.

File: monkey/monkey_island/cc/ui/src/components/ui-components/EventsTable.js [hunk 4]
```diff
@@ -34,21 +32,113 @@ const table_options = {
   selectableRows: 'none'
 };
 
+const timestamp_options =  [{year: 'numeric'}, {month: '2-digit'},{day: '2-digit'},{'hour': '2-digit'},{'minutes': '2-digit'},{'second': 'numeric'}];
+
+const renderTime = (val) => new Date(val*1000).toLocaleString('en-us', timestamp_options);
+
+const renderTarget = (event_target, event_source, machines) => {
+  // event_target is null
+  if ((event_target === null) || (event_target === event_source)) {
```
[the remark this answers is not in the document]
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/ui/src/components/ui-components/EventsTable.js:41]: The rendered one can. They can be hostname or ip which can be the same. I will give it a look.

[the remark this answers is not in the document]
Comment (mssalvatore) [reply monkey/monkey_island/cc/ui/src/components/ui-components/EventsTable.js:41]: I don't think the rendered event target gets passed in to this function.

[the remark this answers is not in the document]
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/ui/src/components/ui-components/EventsTable.js:41]: Fixed!

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**